eng/native/build-commons.sh

```diff
@@ -84,6 +84,8 @@ build_native()
             exit 1
         fi
 
+        cmakeArgs="-C $__RepoRootDir/eng/native/tryrun.cmake $cmakeArgs"
```

Comment:
why is this necessary?

Reply:
Yes, what I meant was why is this not necessary for the mono-based Android build today?

I looked a bit deeper now and I think it is because we're not using any of these cmake checks in the System.Native build so we don't hit this, e.g. we explicitly skip the checks that would cause binaries to be run during configure on mobile and manually set the results: 
https://github.com/dotnet/runtime/blob/c5256458235e00c4bc4634c3cc389fefdd4c10d8/src/native/libs/configure.cmake#L572-L605

So this is fine since we'll override the values for e.g. `HAVE_SHM_OPEN_THAT_WORKS_WELL_ENOUGH_WITH_MMAP` but we should probably clean this up and move all of the detection into tryrun.cmake. Would you mind filing an issue for that?

Reply:
Filed #86829.